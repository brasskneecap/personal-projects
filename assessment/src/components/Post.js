import PostHeader from "./PostHeader"
import PostContent from "./PostContent";
import PostInteraction from "./PostInteraction";
import picture from '../assets/Terran.jpg';
import AddComment from "./AddComment";

const Post = () => {
    let content = "Here's a story all about how my life got flipped turned upside down. I'd like to take a minute just sit right there all tell you all about how I slapped Chris Rock and kept my wifes name out of his mouth"

    return (
        <div>
            <PostHeader picture={picture} headerType="post-header"/>
            <PostContent content={content}/>
            <PostInteraction />
            <AddComment />
        </div>
    )
}

export default Post