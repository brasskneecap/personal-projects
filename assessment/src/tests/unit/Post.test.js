import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Post from '../../components/Post';

describe('Post no comments', () => {
    const wrapper = new ShallowRenderer()

    let post = {
        id: 1,
        userOwner: {
            name: "Test User",
        },
        content: "Test message", 
        reactions: {
            hypes: 0,
            comments: 0,
            shares: 0,
            views: 0
        },
        comments: []
    }

    wrapper.render(
    <Post post={post} />
    );
    
    let componentRendered = wrapper.getRenderOutput();
    
    it('Post Snapshot', () => {
        expect(componentRendered).toMatchSnapshot();
    })

    it('Should contain the passed in content', () => {
        expect(componentRendered.props.children[4].length).toBe(0)
    })
    
})

describe('Post with comments', () => {
    const wrapper = new ShallowRenderer()

    let post = {
        id: 1,
        userOwner: {
            name: "Test User",
        },
        content: "Test message", 
        reactions: {
            hypes: 0,
            comments: 0,
            shares: 0,
            views: 0
        },
        comments: [
            {
                id: 1
            },
            {
                id: 2
            }
        ]
    }

    wrapper.render(
    <Post post={post} />
    );
    
    let componentRendered = wrapper.getRenderOutput();
    
    it('Post Snapshot', () => {
        expect(componentRendered).toMatchSnapshot();
    })

    it('Should contain the passed in content', () => {
        expect(componentRendered.props.children[4].length).toBe(2)
    })
    
})