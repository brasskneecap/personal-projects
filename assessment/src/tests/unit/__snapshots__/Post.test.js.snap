// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post no comments Post Snapshot 1`] = `
<div
  className="base-container"
>
  <PostHeader
    headerType="post-header"
    post={
      Object {
        "comments": Array [],
        "content": "Test message",
        "id": 1,
        "reactions": Object {
          "comments": 0,
          "hypes": 0,
          "shares": 0,
          "views": 0,
        },
        "userOwner": Object {
          "name": "Test User",
        },
      }
    }
  />
  <PostContent
    content="Test message"
  />
  <PostInteraction
    postId={1}
    reactions={
      Object {
        "comments": 0,
        "hypes": 0,
        "shares": 0,
        "views": 0,
      }
    }
  />
  <AddComment
    postId={1}
    type="post"
  />
</div>
`;

exports[`Post with comments Post Snapshot 1`] = `
<div
  className="base-container"
>
  <PostHeader
    headerType="post-header"
    post={
      Object {
        "comments": Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
        ],
        "content": "Test message",
        "id": 1,
        "reactions": Object {
          "comments": 0,
          "hypes": 0,
          "shares": 0,
          "views": 0,
        },
        "userOwner": Object {
          "name": "Test User",
        },
      }
    }
  />
  <PostContent
    content="Test message"
  />
  <PostInteraction
    postId={1}
    reactions={
      Object {
        "comments": 0,
        "hypes": 0,
        "shares": 0,
        "views": 0,
      }
    }
  />
  <AddComment
    postId={1}
    type="post"
  />
  <Comment
    comment={
      Object {
        "id": 1,
      }
    }
    postId={1}
  />
  <Comment
    comment={
      Object {
        "id": 2,
      }
    }
    postId={1}
  />
</div>
`;
